contrib/completion/zsh/_docker

```diff
@@ -138,6 +138,14 @@ __docker_containers_names() {
     __docker_get_containers all names "$@"
 }
 
+__docker_plugins() {
+    [[ $PREFIX = -* ]] && return 1
+    integer ret=1
+    plugins=(${(f)"$(_call_program commands docker $docker_options info | sed -n "/^Plugins/,/^[^ ]/s/ $1: //p")"})
```

Comment:
```
emulate -L zsh
setopt extendedglob
local -a plugins
...
plugins=(${(ps: :)${(M)${(f)${${"$(_call_program commands docker $docker_options info)"##*Plugins:}%%$'\n'^ *}}:# $1: *}## $1: })
```

`${x##*Plugins:}` remove everything before `Plugins`. `${x%%$'\n'^ *}` removes everything that doesn't start by a space. `${(f)x}` turns the remaining into an array. `${(M)x:# Volume: *}` only keeps the line starting with `Volume`, and `${x## Volume: }` remove `Volume`, `${(ps: :)x}` splits on space.

The only drawback is that if there is no "Plugins" section, it will take the first section instead, notably "Containers". As there is no match for Volume/Network and stuff like that, I think this is safe.